pkg/controller/memberstatus/memberstatus_controller_test.go

```diff
@@ -0,0 +1,241 @@
+package memberstatus
+
+import (
+	"testing"
+
+	"github.com/codeready-toolchain/api/pkg/apis"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	. "github.com/codeready-toolchain/member-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var requeueResult = reconcile.Result{RequeueAfter: defaultRequeueTime}
+
+const defaultMemberOperatorName = "member-operator"
+
+func TestNoMemberStatusFound(t *testing.T) {
+	t.Run("No memberstatus resource found", func(t *testing.T) {
+		// given
+		requestName := "bad-name"
+		getHostClusterFunc := newGetHostClusterReady
+		reconciler, req, _ := prepareReconcile(t, requestName, getHostClusterFunc)
+
+		// when
+		res, err := reconciler.Reconcile(req)
+
+		// then - there should not be any error, the controller should only log that the resource was not found
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+	})
+}
+
+func TestOverallStatusCondition(t *testing.T) {
```

Comment:
What about some tests to test the actual `hostConnectionHandleStatus()` and `memberOperatorHandleStatus()`? They are not covered by the tests if I'm not missing anything...
You can test the host one one by mocking `getHostCluster` function and the member one by using a fake client and overriding the `Get()` function to return the expected deployment or an error.

Reply:
The tests were covering most of `hostConnectionHandleStatus()` and `memberOperatorHandleStatus()` but I added a few more cases to get complete coverage. They now have complete coverage:

![image](https://user-images.githubusercontent.com/20015929/85791964-b1420b80-b700-11ea-82d6-ce89a23e10fa.png)

Reply:
Coverage is good now. BTW you can also check the coverage here: https://codecov.io/gh/codeready-toolchain/member-operator/pull/174/diff

Reply:
Ah, thanks! I didn't know it updates before the e2e test completes.